Add openhab2 with reverse proxy, remove auto letsencrypt

hestela opened the pull request:
Main issue i have is that when you try to unpack or cp the files into
the user's home directory the permissions get screwed up. chown -R didn't
work. For now I just switched to the openhab user and unpacked the
pre-built files.

Openhab2 does not support authentication at the moment so I am using
nginx basic auth. Considering adding fail2ban to monitor the nginx auth
logs. Seems that the habmin interface keeps asking for auth repeatedly.

Need to add a crontab entry to run certbot renew

will keep conf directory files here:
[hestela/openhab-conf](https://github.com/hestela/openhab-conf)

Commit:
Add openhab2 with reverse proxy, remove auto letsencrypt

Main issue i have is that when you try to unpack or cp the files into
the user's himdirecotry the permissions get screwed up. chown -R didn't
work. For now I just switched to the openhab user and unpacked the
pre-built files.

Openhab2 does not support authentication at the moment so I am using
nginx basic auth. Considering adding fail2ban to monitor the nginx auth
logs. Seems that the habmin interface keeps asking for auth repeatedly.

Need to add a crontab entry to run certbot renew

hestela commented on nix/openhab.nix:
```diff
@@ -0,0 +1,39 @@
+
+{
+  users.extraUsers.openhab = {
+    isNormalUser = true;
```
i should revisit this, documentation says it should be a system user

hestela commented on nix/openhab.nix:
```diff
@@ -0,0 +1,39 @@
+    after = [ "syslog.target" "network.target" ];
+    environment.HOME = "/home/openhab";
+    environment.USER = "openhab";
+    # TODO: fix manual unpack of zip to home
```
the nix builder cant extract to openhab's homedir

hestela commented:
@acmcarther, idk how interested you are at smart home stuff. but this is automation for connected devices.  my co-worker uses it to schedule things like his sprinkler system, lights. z-wave motions sensors to run scripts, etc. openhab v2 if you are interested. the runtime is java, started by a /bin/bash script. i have anix expression for getting the zip, had some wierd problems i think i know what i could try to get it to work.

acmcarther commented:
ACK:

I have [homeassistant configs](https://github.com/acmcarther/essentials/blob/master/nix/custom-nix-pkgs/homeassistant/default.nix) on my own box, but I've turned em off because keeping the dependencies up to date has been a royal PITA.

acmcarther commented on nix/server.nix:
```diff
@@ -117,8 +118,8 @@ in
     hostId = "e39841f0";
-    firewall.allowedTCPPorts = [ 80 443 3000 8888 42063 8000 ];
-    firewall.allowedUDPPorts = [ 80 443 3000 8888 42063 8000 ];
+    firewall.allowedTCPPorts = [ 80 443 3000 8888 42063 8000 8080 ];
```
Is exposing this port actually necessary? nginx should be proxying from 80/443....

hestela replied:
For LAN access. one of the UIs they have doesnt work with nginx auth, no one has bothered to fix it....

acmcarther commented on nix/development.nix:
```diff
@@ -51,6 +51,7 @@
       go
       go2nix
       godep
+      jre
```
You shouldn't need this dependency here, unless you actually want it as a system dep. Adding it under openhab2's config should work.

hestela replied:
oh yeah, thats right. forgot i had it in the systemd unit

acmcarther commented on nix/nginx.nix:
```diff
@@ -103,6 +90,35 @@ in
+            listen 443;
+            server_name oh2.easycashmoney.org;
+            auth_basic "OpenHab2";
+            auth_basic_user_file /var/www/.htpasswd;
```
Neat.

Commit:
Add mosquitto mqtt broker for openhab

Commit:
Move openhab and ovpn expressions to services dir

Commit:
Fix stop on openhab systemd unit

Commit:
Remove mqtt server, add fauxmo service

mqtt server no longer needed, the new fauxmo service can make a POST
request to the openhab REST API to change item state.

Patches added to disable homeassistant code paths because I won't be
using homeassistant and the homeassistant dependency was too hard to
add.

Fauxmo requires the UDP port 1900 to be open for UPnP broadcast
discovery from the Amazon Echo. And then a port needs to be added for
each Fauxmo device. Fauxmo pretends to be a Wemo smart switch and allows
On/Off commands.

This approach is a lot more responsive than using the built-in Echo
Philips Hue commands.

Commit:
Remove extra ports

Commit:
Fix typo in unit

Commit:
Break out some common options into nix variables in nginx.nix
